cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/SparkRuntimeContextProvider.java

```diff
@@ -262,7 +262,7 @@ private static PluginInstantiator createPluginInstantiator(CConfiguration cConf,
   }
 
   private static Injector createInjector(CConfiguration cConf, Configuration hConf) {
-    return Guice.createInjector(new DistributedProgramRunnableModule(cConf, hConf).createModule());
+    return Guice.createInjector(new DistributedProgramRunnableModule(cConf, hConf).createModule(null, null));
```

Comment:
Similar to MapReduce, if a Spark program adds a partition, we must pass on the program id to make sure it is impersonated correctly.